frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -1304,16 +1314,112 @@ typeConstructorInitialQuery(Context* context, const Type* t)
     CHPL_ASSERT(false && "case not handled");
   }
 
+  // If we're actually running the body of this current query, then the
+  // BuilderResult from any past revision needs to be overwritten.
+
+  auto parentMod = parsing::idToParentModule(context, id);
+  auto modName = "chpl__internal_" + parentMod.symbolName(context).str() + "_" + name.str();
+  auto builder = Builder::createForGeneratedCode(context, modName.c_str(), parentMod.symbolPath());
+  auto dummyLoc = parsing::locateId(context, id);
+
+  // Add a top-level 'use' of the module containing the type being constructed.
+  // This exists to support fields that reference other types in that module.
+  {
+    auto useName = parentMod.symbolName(context);
+    auto ident = Identifier::build(builder.get(), dummyLoc, useName);
+    auto vis = VisibilityClause::build(builder.get(), dummyLoc, std::move(ident));
+
+    AstList alist;
+    alist.push_back(std::move(vis));
+    auto useStmt = Use::build(builder.get(), dummyLoc,
+                              Decl::DEFAULT_VISIBILITY, std::move(alist));
+    builder->addToplevelExpression(std::move(useStmt));
+  }
+
+  // Build formals of type constructor
+  AstList formalAst;
+  for(auto& arg : fieldAsts) {
+    auto var = arg->toVariable();
+    owned<AttributeGroup> attr;
+    if (var->attributeGroup()) {
+      auto attrCopy = var->attributeGroup()->copy();
+      AttributeGroup* asdf = attrCopy.release()->toAttributeGroup();
+      attr = std::make_unique<AttributeGroup>(*asdf);
```

Comment:
`adfg` is almost certainly a debugging name. But also, what's going on here? I'm seeing an invocation of the copy initializer (via `make_unique`) of `AttributeGroup` on top of something that uses `copy`? That seems unnecessary.

Reply:
Whoops! I meant to remove that.